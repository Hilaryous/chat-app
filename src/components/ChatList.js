// @flow
import React from 'react'
import map from 'ramda/src/map'
import AppContext from '../context'
import Chat from './Chat'
import type ChatType from '../types/Chat'

type Props = {
  list: Array<ChatType>,
  selectChat: () => void,
}

const ChatList = ({ selectChat, list }: Props) => (
  <div>
    { map(item => <Chat key={item.toUser} chat={item} selectChat={selectChat} />, list)}
  </div>
)

export default () => (
  <AppContext.Consumer>
    {({ actions: { selectChat }, chats }) => <ChatList selectChat={selectChat} list={chats} />}
  </AppContext.Consumer>
)

// --------------------------
