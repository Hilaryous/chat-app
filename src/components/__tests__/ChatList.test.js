import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import ChatList from '../ChatList'
import AppContext from '../../context'

describe('ChatList', () => {
  it('shows the correct snapshot tree for the ChatList component', () => {
    const tree = renderer.create(
      <StaticRouter location="someLocation" context={{}}>
        <AppContext.Provider>
          <ChatList />
        </AppContext.Provider>
      </StaticRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
