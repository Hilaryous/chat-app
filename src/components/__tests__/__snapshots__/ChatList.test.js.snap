// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList shows the correct snapshot tree for the ChatList component 1`] = `<div />`;
